clippy_utils/src/check_proc_macro.rs

```diff
@@ -123,16 +123,14 @@ fn qpath_search_pat(path: &QPath<'_>) -> (Pat, Pat) {
 
 fn path_search_pat(path: &Path<'_>) -> (Pat, Pat) {
```

Comment:
Thankfully we can't encounter those in most places since they can only be generic constraints. I'll be rewriting the proc macro detection anyways since the current approach isn't great.